src/Revision.php

```diff
@@ -0,0 +1,135 @@
+     * GetMany returns the attributes with keys in $keys from the attribute
+     * table.
+     *
+     * The returned attributes are guaranteed to be correct as of
+     * Reference::revision().
+     *
+     * Non-existent attributes are equivalent to empty attributes, therefore it
+     * is not an error to request keys that have never been created.
+     *
+     * Peers do not always have a copy of the complete attribute table. If any
+     * of the attribute values are unknown they are fetched from the owning peer.
+     *
+     * If any of the attributes can not be retreived because they have already
+     * been modified, StaleFetchException will be thrown. To fetch the attribute
+     * values at the later revision, first call refresh() then retry the
+     * getMany() on the newer revision.
+     *
+     * @return AttributeTable With the the keys $keys.
+     *
+     * @throws StaleFetchException If the attributes were not the latest revision.
+     * @throws NotFoundException   If the session has been closed and the revision can not be queried.
+     */
+    public function getMany(Context $context, $keys): AttributeTable;
+
+    /**
+     * Update atomically modifies the attribute table.
+     *
+     * A successful update produces a new revision.
+     *
+     * Each update is atomic; either all of the attributes in attrs are updated,
+     * or the attribute table remains unchanged. On success, revision is the
+     * newly created revision.
+     *
+     * The following conditions must be met for an update to succeed:
+     *
+     * 1. The session revision represented by this instance must be the latest
+     *    revision. If Reference::revision() is not the latest revision the
+     *    update fails; and StaleUpdateException is thrown.
+     *
+     * 2. All attribute changes must reference non-frozen attributes. If any of
+     *    attributes being updated are already frozen the update fails; and
+     *    FrozenAttributeException is thrown.
+     *
+     * If attrs is empty no update occurs, rev is this revision and err is nil.
+     *
+     * As a convenience, if the update fails for any reason, rev is this
+     * revision. This allows the caller to assign the return value to an
+     * existing variable without first checking for errors.
+     *
+     * @return Revision Latest revision of the session.
+     *
+     * @throws StaleUpdateException When updating any but the latest revision.
+     * @throws NotFoundException    If the session has been closed and the revision can not be queried.
+     */
+    public function update(Context $context, $attrs): Revision;
+
+    /**
+     * Destroy terminates the session.
+     *
+     * The session revision represented by this instance must be the latest
+     * revision. If Reference::revision() is not the latest revision the destroy
+     * fails; and StaleUpdateException is thrown.
+     *
+     * @throws StaleFetchException If the attributes were not the latest revision.
```

Comment:
Destroy is a modification, it'll throw a `StaleUpdateException` just like `update()`